### ISSUE TITLE
Hide Default Group No Resources from Machines section (and column headings on remote machines)

### ISSUE THREAD

**Issue by muziqaz** [FoldingAtHome/fah-web-client-bastet#192]
![resource groups](https://github.com/user-attachments/assets/e78cfb40-7d53-43e5-a8e2-c70de69e3bda)

I know we spoke about this before, that disabling Default group is a bit of the undertaking. Would it be possible to just hide it from Machines section Default Group if it does not have any assigned resources to it? At the moment UI gets quite busy, and that extra line per machine with that triangle at the end gets very distracting if one is monitoring their machines. Plus it would save some real estate vertically, so more remote machines would be visible without the need of scrolling down.

And maybe even hide column headings (Status, Project, RCG, Core, etc) for all but local machine. Here is an example of what I mean:
![resource groups compact](https://github.com/user-attachments/assets/7bf1a926-006a-4bae-9ed0-e72876653b4b)

So now if I glance at the bottom screenshot, I can easily see all the groups and machines are folding, and none are paused, while the first screenshot would make me double check, since I would spot these "Start" triangles at the end.

Thanks

**Comment by kbernhagen** [comment]
If those are separate tables for each machine, the columns might not line up because of variable width content.

**Comment by muziqaz** [comment]
Yeah, I noticed that, though this might be mitigated by limiting character number in Machine name.
Finishing and Running length differences are not too bad, though. But overall, a lot of columns might have different lengths, you have a point.
This issue writeup came about mainly as to hide Default Resource Group, and column headings were just a "oh, by the way, this can go away" afterthought.

**Comment by kbernhagen** [comment]
Might be nice to have an option to hide *all* groups with no resources, unless that would leave none visible.

**Comment by muziqaz** [comment]
I think just a Default Group with no resources should be hidden, as other groups might have no resources temporary while user is setting things up, and that might cause a lot of questions for new users.

**Comment by muziqaz** [comment]
I will just throw this in here :D 
![ideas ideas](https://github.com/user-attachments/assets/ecbf59ce-01de-4723-be68-3e91ae493e39)

**Comment by kbernhagen** [comment]
Seems reasonable. If you have a wide screen.
The action buttons would only apply to the entire group.
Might confuse people who think they can pause individual WUs.

**Comment by jcoffland** [comment]
I think it makes sense to not require the default group.  Internally, the default group has the empty string as its name and it cannot be deleted.  If you could rename the default group and delete it just like any other group then this wouldn't be an issue.

The other issue of combining the machine and groups sections in to one table makes some sense.  As @kbernhagen mentioned, it leaves no place for the group or machine action buttons.

Finally, we have to strike a balance between the dense table view expert folders might like and a more user friendly view that makes F@H less daunting for new users.  And it still needs to be usable on mobile screens.

**Comment by muziqaz** [comment]
Would it be possible to add colour coding for different machines? To give more contrast where one machine starts and another ends?
Above suggestions where just ideas, I understand that from my point of view with wide or ultra wide monitors, it might seem like we have loads of available space, but that wouldn't be the case for 4:3 crowd.
Though I would appreciate if `Default Group|No Resources` thing got hidden

**Comment by kbernhagen** [comment]
Currently units have a group name, but that group is not guaranteed to exist. If a unit’s group has been deleted, the unit seems to belong to the default group. 

If you remove group “” and a user has multiple groups created, which of those groups should be treated as the default group when a unit must be moved?

**Comment by kbernhagen** [comment]
Might also want to disable deleting a group that has units or is not paused.

**Comment by kbernhagen** [comment]
Note that a GPU unit that gets moved to a group with no appropriate GPU will be stuck paused for insufficient resources.

**Comment by muziqaz** [comment]
I think just hiding it from main page would be sufficient. As far as I understand, removing Default Group is out of the question due to technical reasons.
So if Default Group has no resources assigned to it, just don't show it on main page. In settings it will still be visible and if one wants to add some resources back to it, it can be unhidden in the main page.

**Comment by jcoffland** [comment]
v8.4.5 add some extra columns: ``Version``, ``Resources`` & ``Group``.  This is a step towards supporting a table with only WUs and no machine or group rows.

I still don't see a good way to display machine and group actions.  One idea would be to have a separate table of machines and groups with no WUs and only show WUs on the WUs tab but then you have to move back and forth.  Another idea is to have a popup with all the machine and group actions.  I'm still not sold on either of these ideas.

**Comment by muziqaz** [comment]
I played around with different resolutions and monitor ratios. While my sketch proposal would work on normal 16:9, 16:10 or wider screens, it fails miserably with 4:3 ratio screens, unless users with such monitors majorly sacrifice amount of info columns they can add to main page without scrolling horizontally.
If we could settle for hidden Default Group | No Resource line, and maybe some colour coding for resource groups, that would probably be sufficient.
In regards to color coding, user could shade their CPU groups one color, while GPU groups something else. Or colour shades could be predetermined differenciating between CPU group and GPU group. If colours are too complicated, this is just an idea, and not really a requirement. Just thinking out loud to add a bit of contrast for eyes to pick up different machines and different groups quicker (but that might come with time, once we get completely used to UI). Plus I would think that web control UI might be open for some clever people writing plugins/skins for v8 down the line.

**Comment by kbernhagen** [comment]
After restructuring lufah internals, I plan to give it an optional TUI using textual. My mockup interface might me interesting as a point of discussion. Here is a screenshot at terminal 80x24:

![image](https://github.com/user-attachments/assets/ae08eeb0-fa85-49d5-acc4-f6e9cf9062ff)
